### ISSUE TITLE
Formal description of vector tiles schema

### ISSUE THREAD

**Issue by jirik** [openmaptiles/openmaptiles#229]
Disclaimer: This is just an idea with lower priority.

It would be useful to have formal description of vector tiles schema. Having this, we can make better check if a GL style is correct (i.e. contains only existing layers, attributes, and values).

Such formal description should contain at least
- list of layers
- list of attributes of each layer
- domain of each attribute

Similar information is right now at each layer's yaml. Question is how detailed and up-to-date it is, because it's updated manually.

**Comment by lukasmartinelli** [comment]
The YAML file is already a formal description.

I don't think you can get more formal than the YAML description without much effort of either always executing queries or parsing SQL ASTs.

Better make sure the YAML file is up to date (it needs to be if you want useful docs) and then you can always collect these YAML files to build one formal description.

**Comment by jirik** [comment]
@lukasmartinelli You are right, YAML files contain a lot of information for the start, but what about data type? Are we able to get field's data type (number, string, boolean) from current yamls?

**Comment by jirik** [comment]
Or what about geometry type?

**Comment by lukasmartinelli** [comment]
You could more metadata for fields decribing its type.
